src/store/modules/swap/ethBancor.ts

```diff
@@ -3197,6 +3197,8 @@ export class EthBancorModule
       )
       .flatMap(relay =>
         relay.reserves.map(reserve => {
+          const viewRelay = this.relay(relay.id);
```

Comment:
I don't think it's a good idea to lookup the pool in this way, might open us up to a race condition as this is dependent on another getter. 

Also we're looping over the relay then the reserves, so such a lookup can be performed just the once rather than twice. 

Lets base ```liquidityProtection``` off whether or not the relay id exists in the ```whitelistedPools``` array instead.

Reply:
> I don't think it's a good idea to lookup the pool in this way, might open us up to a race condition as this is dependent on another getter.
> 
> Also we're looping over the relay then the reserves, so such a lookup can be performed just the once rather than twice.
> 
> Lets base `liquidityProtection` off whether or not the relay id exists in the `whitelistedPools` array instead.

added, thanks for flagging